packages/core/database/src/schema/diff.ts

```diff
@@ -194,30 +211,37 @@ export default (db: Database) => {
     };
   };
 
-  const diffTableColumns = (srcTable: Table, destTable: Table): ColumnsDiff => {
+  const diffTableColumns = (diffCtx: TableDiffContext): ColumnsDiff => {
+    const { databaseTable, userSchemaTable, previousTable } = diffCtx;
+
     const addedColumns: Column[] = [];
     const updatedColumns: ColumnDiff['diff'][] = [];
     const unchangedColumns: Column[] = [];
     const removedColumns: Column[] = [];
 
-    for (const destColumn of destTable.columns) {
-      const srcColumn = helpers.findColumn(srcTable, destColumn.name);
-      if (srcColumn) {
-        const { status, diff } = diffColumns(srcColumn, destColumn);
+    for (const userSchemaColumn of userSchemaTable.columns) {
+      const databaseColumn = helpers.findColumn(databaseTable, userSchemaColumn.name);
+
+      if (databaseColumn) {
+        const { status, diff } = diffColumns(databaseColumn, userSchemaColumn);
 
         if (status === statuses.CHANGED) {
           updatedColumns.push(diff);
         } else {
-          unchangedColumns.push(srcColumn);
+          unchangedColumns.push(databaseColumn);
         }
       } else {
-        addedColumns.push(destColumn);
+        addedColumns.push(userSchemaColumn);
       }
     }
 
-    for (const srcColumn of srcTable.columns) {
-      if (!helpers.hasColumn(destTable, srcColumn.name)) {
-        removedColumns.push(srcColumn);
+    for (const databaseColumn of databaseTable.columns) {
+      if (
+        !helpers.hasColumn(userSchemaTable, databaseColumn.name) &&
+        previousTable &&
```

Comment:
Doesn't that mean that whatever happens, if `previousTable` is undefined, it won't enter this condition?

Is it on purpose?

Reply:
yes exactly, the idea is that if the column is in db and was never tracked by us (so not in a stored schema) we should not delete it. as soon as that column gets into a schema it will start being tracked and deleted if removed